File: src/test/java/engineer/thomas_werner/euler/Problem12Test.java
package engineer.thomas_werner.euler;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.math.BigInteger;

public class Problem12Test {

    @Test
    public void testGetDivisorCount() {
        final Problem12 problem12 = new Problem12();
        Assertions.assertEquals(1, problem12.getDivisorCount(BigInteger.ONE));
        Assertions.assertEquals(2, problem12.getDivisorCount(BigInteger.valueOf(3)));
        Assertions.assertEquals(4, problem12.getDivisorCount(BigInteger.valueOf(6)));
        Assertions.assertEquals(4, problem12.getDivisorCount(BigInteger.valueOf(10)));
        Assertions.assertEquals(4, problem12.getDivisorCount(BigInteger.valueOf(15)));
        Assertions.assertEquals(4, problem12.getDivisorCount(BigInteger.valueOf(21)));
        Assertions.assertEquals(6, problem12.getDivisorCount(BigInteger.valueOf(28)));
    }

}

File: src/test/kotlin/de/huddeldaddel/euler/Problem12Test.kt
package de.huddeldaddel.euler

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

import java.math.BigInteger

class Problem12Test {

    @Test
    fun testGetDivisorCount() {
        Assertions.assertEquals(1, Problem12.getDivisorCount(BigInteger.ONE))
        Assertions.assertEquals(2, Problem12.getDivisorCount(BigInteger.valueOf(3)))
        Assertions.assertEquals(4, Problem12.getDivisorCount(BigInteger.valueOf(6)))
        Assertions.assertEquals(4, Problem12.getDivisorCount(BigInteger.valueOf(10)))
        Assertions.assertEquals(4, Problem12.getDivisorCount(BigInteger.valueOf(15)))
        Assertions.assertEquals(4, Problem12.getDivisorCount(BigInteger.valueOf(21)))
        Assertions.assertEquals(6, Problem12.getDivisorCount(BigInteger.valueOf(28)))
    }

}
